component/src/test/java/org/wso2/extension/siddhi/io/websocket/sink/util/WebSocketReceiver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) 2017 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.extension.siddhi.io.websocket.sink.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.transport.http.netty.contract.websocket.HandshakeFuture;
+import org.wso2.transport.http.netty.contract.websocket.HandshakeListener;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketClientConnector;
+import org.wso2.transport.http.netty.contract.websocket.WsClientConnectorConfig;
+import org.wso2.transport.http.netty.contractimpl.HttpWsConnectorFactoryImpl;
+
+import javax.websocket.Session;
+
+public class WebSocketReceiver {
+    private static Logger log = LoggerFactory.getLogger(WebSocketReceiver.class);
+
+    public WebSocketReceiver(String url, ResultContainer resultContainer) {
+        HttpWsConnectorFactoryImpl httpConnectorFactory = new HttpWsConnectorFactoryImpl();
+        WsClientConnectorConfig configuration = new WsClientConnectorConfig(url);
+        WebSocketClientConnector clientConnector = httpConnectorFactory.createWsClientConnector(configuration);
+        WebSocketClientConnectorListener connectorListener = new WebSocketClientConnectorListener();
+        HandshakeFuture handshakeFuture = clientConnector.connect(connectorListener);
+        handshakeFuture.setHandshakeListener(new HandshakeListener() {
```

Comment:
Why anonymous class? AFAIR, we discussed this in the code review and agreed to move into a "named" (either inner of separate) class. 

**Please attend to all comments/feedback given in the code review.**

Reply:
fixed in 9fea7d3